### PR TITLE
First iteration of Marvel API avatar choosing

### PR DESCRIPTION
**Pull request by CalexisGonzalez**
First iteration with recyclerview holding cardviews, and imagedialog pop up.

### CODE REVIEW

File: app/src/main/java/com/example/fittracker/adapter/MarvelAdapter.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    public PhotoViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        final View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.cardview_avatar,
+                parent, false);
+        return new PhotoViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(final PhotoViewHolder viewHolder, int position) {
+        final Result imageInfo = items.get(position);
+        String imageUrl = imageInfo.getThumbnail().getPath() + "." + imageInfo.getThumbnail().getExtension();
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/adapter/MarvelAdapter.java]: remove hardcode pls

File: app/src/main/java/com/example/fittracker/mvp/view/AvatarListView.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    RecyclerView recyclerView;
+    @BindView(R.id.avatarlist_progress_bar)
+    ProgressBar progressBar;
+
+    public AvatarListView(Activity activity) {
+        this.activity = new WeakReference<>(activity);
+        ButterKnife.bind(this, activity);
+        recyclerView.setAdapter(new MarvelAdapter(new ArrayList<Result>()));
+        LinearLayoutManager llm = new LinearLayoutManager(this.activity.get());
+        recyclerView.setLayoutManager(llm);
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/mvp/view/AvatarListView.java:37]: put this last three lines in a separate method. Something like initAdapter

File: app/src/main/res/layout/cardview_avatar.xml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+            <RelativeLayout
+                android:id="@+id/avatar_recyclerview_image_layout"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_alignParentStart="true"
+                android:layout_centerVertical="true"
+                android:paddingEnd="@dimen/avatar_recyclerview_padding">
+
+
```
Comment (SartiMau) [inline app/src/main/res/layout/cardview_avatar.xml]: remove unnecessary extra line

File: app/src/main/res/layout/cardview_avatar.xml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                    android:layout_gravity="center" />
+
+                <ImageView
+                    android:id="@+id/avatar_recyclerview_image"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content" />
+
+            </RelativeLayout>
+
+
```
Comment (SartiMau) [inline app/src/main/res/layout/cardview_avatar.xml]: remove unnecessary extra line

File: app/src/main/res/layout/dialog_avatar_image.xml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+
+    <RelativeLayout
+        android:id="@+id/dialog_avatar_layout"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content">
+
+
```
Comment (SartiMau) [inline app/src/main/res/layout/dialog_avatar_image.xml]: remove unnecessary extra line

File: app/src/main/res/layout/dialog_avatar_image.xml [hunk 6]
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+
```
Comment (SartiMau) [inline app/src/main/res/layout/dialog_avatar_image.xml]: remove unnecessary extra line

File: app/src/main/java/com/example/fittracker/dialog/MarvelDialog.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+import com.example.fittracker.ConstantUtils;
+import com.example.fittracker.R;
+import com.squareup.picasso.Callback;
+import com.squareup.picasso.Picasso;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+public class MarvelDialog extends Dialog {
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/dialog/MarvelDialog.java:16]: use MVP for this pls

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/fittracker/adapter/MarvelAdapter.java (hunk 1 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/adapter/MarvelAdapter.java]: Done

File: app/src/main/java/com/example/fittracker/mvp/view/AvatarListView.java (hunk 2 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/mvp/view/AvatarListView.java:37]: Done

File: app/src/main/res/layout/cardview_avatar.xml (hunk 3 above)
Comment (CalexisGonzalez) [reply app/src/main/res/layout/cardview_avatar.xml]: Done

File: app/src/main/res/layout/cardview_avatar.xml (hunk 4 above)
Comment (CalexisGonzalez) [reply app/src/main/res/layout/cardview_avatar.xml]: Done

File: app/src/main/res/layout/dialog_avatar_image.xml (hunk 5 above)
Comment (CalexisGonzalez) [reply app/src/main/res/layout/dialog_avatar_image.xml]: Done

File: app/src/main/res/layout/dialog_avatar_image.xml (hunk 6 above)
Comment (CalexisGonzalez) [reply app/src/main/res/layout/dialog_avatar_image.xml]: Done

File: app/src/main/java/com/example/fittracker/dialog/MarvelDialog.java [hunk 8]
```diff
@@ -35,26 +29,12 @@ protected void onCreate(Bundle savedInstanceState) {
         ButterKnife.bind(this);
         setTitle(null);
         setCancelable(false);
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/dialog/MarvelDialog.java]: can you put these two methods in the init method of the view?

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/fittracker/dialog/MarvelDialog.java (hunk 8 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/dialog/MarvelDialog.java]: Done

**Review by SartiMau — APPROVED** [review]

File: app/src/main/java/com/example/fittracker/adapter/MarvelAdapter.java [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+            @Override
+            public void onSuccess() {
+                if (viewHolder.progressBar != null) {
+                    viewHolder.progressBar.setVisibility(View.GONE);
+                }
+            }
+
+            @Override
+            public void onError(Exception e) {
+
```
Comment (priettt) [inline app/src/main/java/com/example/fittracker/adapter/MarvelAdapter.java]: You could show a toast here, so the user knows something went wrong.

File: app/src/main/java/com/example/fittracker/mvp/presenter/AvatarListPresenter.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+        call.enqueue(new Callback<MarvelPojo>() {
+            @Override
+            public void onResponse(Call<MarvelPojo> call, Response<MarvelPojo> response) {
+                results = response.body().getData().getResults();
+                view.setAdapter(new MarvelAdapter(results));
+            }
+
+            @Override
+            public void onFailure(Call<MarvelPojo> call, Throwable t) {
+                return;
```
Comment (priettt) [inline app/src/main/java/com/example/fittracker/mvp/presenter/AvatarListPresenter.java]: Same here

File: app/src/main/java/com/example/fittracker/mvp/presenter/MarvelDialogPresenter.java [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+        Picasso.get().load(imageUrl).resize(ConstantUtils.MARVEL_IMAGE_LARGE_DIMENSION,
+                ConstantUtils.MARVEL_IMAGE_LARGE_DIMENSION).into(view.getImageView(), new Callback() {
+            @Override
+            public void onSuccess() {
+                view.hideProgressBar();
+            }
+
+            @Override
+            public void onError(Exception e) {
+
```
Comment (priettt) [inline app/src/main/java/com/example/fittracker/mvp/presenter/MarvelDialogPresenter.java]: same here

File: app/build.gradle [hunk 12]
```diff
@@ -14,6 +14,11 @@ android {
         buildConfigField "String", "mail_host", "\"smtp.gmail.com\""
         buildConfigField "String", "weather_api_key", "\"852c6dcc0bd878109bd0fc7bb1566c3e\""
         buildConfigField "String", "weather_icon_url", "\"https://openweathermap.org/img/w/\""
+        buildConfigField "String", "marvel_api_key_public", "\"08cd7f2cd10462f4d20ab91d76ecec33\""
```
Comment (priettt) [inline app/build.gradle]: All the constants here should be in UPPER_CASE.

**Review by priettt — CHANGES_REQUESTED** [review]
Just some little comments, everything looks fine overall.

Also, you should do some code reformatting (CTRL+ALT+L). Code style is wrong in some classes, like AvatarListActivity.

File: app/build.gradle (hunk 12 above)
Comment (CalexisGonzalez) [reply app/build.gradle]: Done

File: app/src/main/java/com/example/fittracker/adapter/MarvelAdapter.java (hunk 9 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/adapter/MarvelAdapter.java]: Done

File: app/src/main/java/com/example/fittracker/mvp/presenter/AvatarListPresenter.java (hunk 10 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/mvp/presenter/AvatarListPresenter.java]: Done

File: app/src/main/java/com/example/fittracker/mvp/presenter/MarvelDialogPresenter.java (hunk 11 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/mvp/presenter/MarvelDialogPresenter.java]: Done

Comment (priettt) [reply app/src/main/java/com/example/fittracker/mvp/presenter/MarvelDialogPresenter.java]: I don't see it solved here. Did you push all the changes?

Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/mvp/presenter/MarvelDialogPresenter.java]: There I solved it

**Review by priettt — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Pull request merged**